src/crimson/os/seastore/cached_extent.h

```diff
@@ -549,12 +549,15 @@ class CachedExtent
   }
 
   bool is_stable_writting() const {
-    // MUTATION_PENDING and under-io extents are already stable and visible,
-    // see prepare_record().
+    // MUTATION_PENDING/INITIAL_WRITE_PENDING and under-io extents are already
+    // stable and visible, see prepare_record().
     //
-    // XXX: It might be good to mark this case as DIRTY from the definition,
+    // XXX: It might be good to mark this case as DIRTY/CLEAN from the definition,
     // which probably can make things simpler.
-    return is_mutation_pending() && is_pending_io();
+    if (is_pending_io()) {
+      assert(is_mutation_pending() || is_initial_pending());
```

Comment:
See https://github.com/ceph/ceph/pull/63389#discussion_r2110723973, I think this seems to be:
`return (is_mutation_pending() || is_initial_pending()) && is_pending_io();`

It shouldn't validate that pending-io must be initial-pending/mutation-pending, right ?